skcosmo/selection/_base.py

```diff
@@ -0,0 +1,96 @@
+from abc import abstractmethod
+
+
+class _BaseSelection:
```

Comment:
I'm not completely sold on using deep class hierarchy here. Does sklearn do this or something similar? 

In Python, class inheritance is mostly useful to provide implementation of common methods, which is not the case here. The only method implemented is `progress`, which is not related to the core of what a "Selection method" is.